### PR TITLE
Ensure that embedded broadcast checks the base type of the parent node

### PR DESCRIPTION
**Pull request by tannergooding**
`IsContainableHWIntrinsicOp` was only checking the base type of the broadcast node, which meant that scenarios such as `MultiplyLow(vector256Int16, Vector256.Create(int32).AsInt16())` would incorrectly try to contain the child node.

This ensures that we check both the base type of the broadcast (which indicates how many bytes it reads from memory) and the base type of the parent node (which indicates how many bytes it would read from memory).

### CODE REVIEW

File: src/coreclr/jit/lowerxarch.cpp [hunk 1]
```diff
@@ -8202,18 +8202,20 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
         case NI_AVX2_BroadcastScalarToVector256:
         case NI_AVX512F_BroadcastScalarToVector512:
         {
-            var_types baseType = hwintrinsic->GetSimdBaseType();
-            if (varTypeIsSmall(baseType))
+            var_types parentBaseType = parentNode->GetSimdBaseType();
+            var_types childBaseType  = hwintrinsic->GetSimdBaseType();
+
+            if (varTypeIsSmall(parentBaseType) || varTypeIsSmall(childBaseType))
```
Comment (jakobbotsch) [inline src/coreclr/jit/lowerxarch.cpp:8208]: Is this sufficient? For example, doesn't
```csharp
[MethodImpl(MethodImplOptions.NoInlining)]
public static Vector256<long> Test(Vector256<long> x, ref int y)
{
    return Avx2.Add(x + x, Vector256.Create(y).AsInt64());
}
```
have the same problem and still pass these checks?

Comment (tannergooding) [reply src/coreclr/jit/lowerxarch.cpp:8208]: Yes, good point. Will do a check that the sizes match here.

There's a "todo" in that it is safe for some constants, but that's a different path and is already handled.

Comment (tannergooding) [reply src/coreclr/jit/lowerxarch.cpp]: Fixed.

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**